### PR TITLE
Intermcu cleanup

### PR DESCRIPTION
**Pull request by fvantienen**
Cleanup the InterMCU and update pprzlink for multiple parser support.

### CODE REVIEW

**Comment by fvantienen** [comment]
https://github.com/paparazzi/pprzlink/pull/10 must be merged before this.

File: sw/airborne/subsystems/intermcu/intermcu_fbw.c [hunk 1]
```diff
@@ -40,48 +41,51 @@ static uint8_t px4RebootSequence[] = {0x41, 0xd7, 0x32, 0x0a, 0x46, 0x39};
...
+INFO("InterMCU UART will only send 8 radio channels!")
 #endif
 
-// Used for communication
-static struct link_device *intermcu_device = (&((INTERMCU_LINK).device));
-static struct pprz_transport intermcu_transport;
+/* Main InterMCU structure */
+struct intermcu_t intermcu = {
+  .device = (&((INTERMCU_LINK).device)),
+  .enabled = TRUE,
```
Comment (flixr) [inline sw/airborne/subsystems/intermcu/intermcu_fbw.c]: should use stdbool `true` and `false`

File: sw/airborne/subsystems/intermcu/intermcu_fbw.c [hunk 2]
```diff
@@ -160,34 +167,50 @@ static void intermcu_parse_msg(struct transport_rx *trans, void (*commands_frame
...
+      for (i = 0; i < intermcu.transport.trans_rx.payload_len; i++) {
+        imcu_msg_buf[i] = intermcu.transport.trans_rx.payload[i];
+      }
+
+      intermcu.msg_available = true;
+      intermcu.transport.trans_rx.msg_received = false;
     }
   }
+
+  if(intermcu.msg_available) {
```
Comment (flixr) [inline sw/airborne/subsystems/intermcu/intermcu_fbw.c]: pleas follow code style (space after if)

**Comment by fvantienen** [comment]
Everything should be fixed now ;)

File: sw/airborne/subsystems/intermcu.h [hunk 3]
```diff
@@ -51,13 +52,18 @@ enum intermcu_PX4_baud_status {
 };
 
 struct intermcu_t {
-  enum intermcu_status status;
-  uint8_t time_since_last_frame;
+  struct link_device *device;       ///< Device used for communication
+  struct pprz_transport transport;  ///< Transport over communication line (PPRZ)
+  enum intermcu_status status;      ///< Status of the INTERMCU
+  uint8_t time_since_last_frame;    ///< Time since last frame
+  uint8_t enabled;                  ///< If the InterMCU communication is enabled
```
Comment (flixr) [inline sw/airborne/subsystems/intermcu.h]: probably also bool?

**Comment by fvantienen** [comment]
Is this ok to merge now?

**Comment by gautierhattenberger** [comment]
I think it is fine now

**Pull request merged**